litex/soc/interconnect/packet.py

```diff
@@ -337,13 +345,14 @@ def __init__(self, sink_description, source_description, header):
             )
         )
         self.sync += If(sink.valid & sink.ready, sink_d.eq(sink))
+        print(header_words, header_leftover)
         fsm.act("UNALIGNED-DATA-COPY",
             source.valid.eq(sink.valid | sink_d.last),
             source.last.eq(sink.last | sink_d.last),
             sink.ready.eq(source.ready),
             source.data.eq(sink_d.data[header_leftover*8:]),
             source.data[min((bytes_per_clk-header_leftover)*8, data_width-1):].eq(sink.data),
-            If(fsm_from_idle,
```

Comment:
@enjoy-digital fsm_from_idle is a misleading name. As this is active here even when the FSM transitions from HEADER-RECEIVE.